### DISCUSSION TITLE
V200R001C00SPC132 firmware in SUN2000L-[X]KTL-L1

### DISCUSSION THREAD

**Discussion by PacmanForever** [wlcrs/huawei_solar#503] [General]
Hello,

has anyone tried the latest inverter firmware version (V200R001C00SPC132) for the SUN2000L-[X]KTL-L1 models?

[X] can be 4, 5, 6 etc

Does this integration work well?

Thanks

**Comment by Roving-Ronin** [comment]
@jvcsw 

The list of firmware / models that people have tested is kept at: https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware#confirmed-firmware-versions

Whilst the M1's are tested up to SPC155.... and v156 just adds some OEM branding for Australia / New Zealand, v157 adds EMMA support and v158 dropped last week and just adds Philipines / Isreali grid support. 

The latest anyone has tested (and updated to this list) is V200R001C00SPC132 for L1's. For M1's its tested to V100R001C00SPC155.

Looking at the release notes for the L1's it looks like SPC133 ( you can freely get from https://photomate.zendesk.com/hc/en-gb/articles/4408009530909-SUN2000-2-6-KTL-L1 ) this is the same as the M1's SPC156.  As this just adds some branding for AU/NZ this should work fine on your L1 + WLCRS.

L1 SPC135 firmware would therefore look to be the same as the M1's SPC158, so if you have access to that you can add support for the forthcoming EMMA (replaces sdongle / DDSU-666 meter / logger). If you have registered you inverter and can access, it can be downloaded from: https://support.huawei.com/enterprise/en/fusionsolar/sun2000l-pid-22027611/software/261182421?idAbsPath=fixnode01%7C258788303%7C254827209%7C258792409%7C22027611

Whatever you do, remember to update the supported firmware list, at the above link.

**Reply by PacmanForever** [reply]
I'm sorry for being late in replying. I needed time to do the tests.

All works fine with V200R001C00SPC133 but I'm using option A and in the table is setted to B.

I have updated the table.

Thank you very much!!

**Reply by Roving-Ronin** [reply]
@jvcsw Thanks for the update. BTW if you did end up looking to use the Option B (sDongle) i've recently updated mine to the latest V200R022C10SPC108 and it works fine.

**Reply by PacmanForever** [reply]
SDongle updated, thank you.
But I can't connect using the option B with my inverter using ports 502 and 6607. Always appears this error:

![imagen](https://github.com/wlcrs/huawei_solar/assets/37216513/0adc2b2c-77a5-489c-98df-51c59428bb57)

**Reply by Roving-Ronin** [reply]
@jvcsw assuming you've enabled Modbus access and set unrestricted access (from local network) in the inverter settings. Have you tried using a program like https://qmodbus.sourceforge.net to connect to the inverter on those ports, confirming it's running or there's a network issue?

**Reply by PacmanForever** [reply]
I have tried to connect to these ports with telnet and it is impossible. The inverter must have them closed through the SDongle.

**Reply by Roving-Ronin** [reply]
@jvcsw  https://forum.huawei.com/enterprise/en/modbus-tcp-guide/thread/667250677153415168-667213868771979264

**Reply by PacmanForever** [reply]
ok, ok, ok.

Checking the fusion solar website I see that the modbus channel id is 3 and not 0 or 1.

![imagen](https://github.com/wlcrs/huawei_solar/assets/37216513/fc8a5e84-be3e-488f-a6d2-d899cf72e0ee)

Now works fine.

Could be interesting change this:

![imagen](https://github.com/wlcrs/huawei_solar/assets/37216513/59e04d58-f173-4cf3-991e-5d8a0caea779)

**Reply by Roving-Ronin** [reply]
Nothing to change in the documentation, as it suggests confirming the numbers. This number is set by the installer when first configuring the inverter, for some strange reason your installer has set it to 3. 
In my case I have 2 on the primary and 1 on the secondary inverter... as set by the approved electrician/ installer. 🤷

**Comment by JensenNick** [comment]
I can confirm that the latest firware ...132 for the ...L1 models is working well with the integration.
![2023-11-25_08-20-16](https://github.com/wlcrs/huawei_solar/assets/149193835/bb6be38e-04b5-4ef2-bcc2-13de7887dd1a)
